Parser/tokenizer.c

```diff
@@ -829,12 +844,25 @@ tok_readline_raw(struct tok_state *tok)
         if (!tok_reserve_buf(tok, BUFSIZ)) {
             return 0;
         }
-        char *line = Py_UniversalNewlineFgets(tok->inp,
-                                              (int)(tok->end - tok->inp),
-                                              tok->fp, NULL);
+        memset(tok->inp, 0, BUFSIZ);
+        int n_chars = (int)(tok->end - tok->inp);
+        char *line = Py_UniversalNewlineFgets(tok->inp, n_chars, tok->fp, NULL);
         if (line == NULL) {
             return 1;
         }
+        
+        // Find the first non null character starting from the right of tok->inp
+        char *last_char = tok->inp + n_chars - 1;
```

Comment:
is a bit unfortunate that we need to do this to know the last character that was ingested, but the alternative is reimplementing `Py_UniversalNewlineFgets` so it tells us how many chars actually added to the buffer

Reply:
I have added a commit adding a new interface for `Py_UniversalNewlineFgets` that gives us the size for consideration